test/examples/ucp_client_server.c

```diff
@@ -0,0 +1,450 @@
+}
+
+/**
+ * The callback on the server side which is invoked upon receiving a connection
+ * request from the client.
+ */
+static void server_accept_cb(ucp_ep_h ep, void *arg)
+{
+    ucx_server_ctx_t *context = arg;
+
+    /* Save the server's endpoint in the user's context, for future usage */
+    context->ep = ep;
+}
+
+/**
+ * Create a listener on the server side to listen on the given address.
+ */
+ucs_status_t server_listen(ucp_worker_h ucp_worker, const struct sockaddr* addr,
+                           socklen_t addrlen, ucp_listener_h *listener,
+                           ucx_server_ctx_t *context)
+{
+    ucp_listener_params_t params;
+    ucs_status_t status;
+
+    params.field_mask         = UCP_LISTENER_PARAM_FIELD_SOCK_ADDR |
+                                UCP_LISTENER_PARAM_FIELD_ACCEPT_HANDLER;
+    params.sockaddr.addr      = addr;
+    params.sockaddr.addrlen   = addrlen;
+    params.accept_handler.cb  = server_accept_cb;
+    params.accept_handler.arg = context;
+
+    status = ucp_listener_create(ucp_worker, &params, listener);
+    if (status == UCS_OK) {
+        printf("Waiting for connection...\n");
+    } else {
+        fprintf(stderr, "failed to listen\n");
+    }
+
+    return status;
+}
+
+/**
+ * Set an address for the server to listen on - INADDR_ANY on a well known port.
+ */
+void set_listen_addr(struct sockaddr_in *listen_addr)
+{
+    /* The server will listen on INADDR_ANY */
+    memset(listen_addr, 0, sizeof(struct sockaddr_in));
+    listen_addr->sin_family      = AF_INET;
+    listen_addr->sin_addr.s_addr = INADDR_ANY;
+    listen_addr->sin_port        = server_port;
+}
+
+/**
+ * Set an address to connect to. A given IP address on a well known port.
+ */
+void set_connect_addr(const char *ip, struct sockaddr_in *connect_addr)
+{
+    memset(connect_addr, 0, sizeof(struct sockaddr_in));
+    connect_addr->sin_family      = AF_INET;
+    connect_addr->sin_addr.s_addr = inet_addr(ip);
+    connect_addr->sin_port        = server_port;
+}
+
+/**
+ * Initialize the server side. The server starts listening on the set address
+ * and waits for its connected endpoint to be created.
+ */
+static int start_server(ucp_worker_h ucp_worker, ucx_server_ctx_t *context,
+                        ucp_listener_h *listener)
+{
+    struct sockaddr_in listen_addr;
+    ucs_status_t status;
+
+    context->ep = NULL;
+
+    set_listen_addr(&listen_addr);
+    status = server_listen(ucp_worker, (const struct sockaddr*)&listen_addr,
+                           sizeof(listen_addr), listener,
+                           context);
+
+    if (status == UCS_OK) {
+        while (context->ep == NULL) {
+            ucp_worker_progress(ucp_worker);
+        }
+    }
+
+    return status;
+}
+
+/**
+ * Initialize the client side. Create an endpoint from the client side to be
+ * connected to the remote server (to the given IP).
+ */
+static ucs_status_t start_client(ucp_worker_h ucp_worker, const char *ip,
```

Comment:
need to be consistent with return codes, with all function will return 'int' or 'ucs_status_t'
i would recommend 'int' because it's example code